src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -87,24 +97,23 @@ private LuceneFieldMapper(
             originalMappingParameters
         );
         KNNMappingConfig knnMappingConfig = mappedFieldType.getKnnMappingConfig();
-        KNNMethodContext knnMethodContext = knnMappingConfig.getKnnMethodContext()
-            .orElseThrow(() -> new IllegalArgumentException("KNN method context is missing"));
+        KNNMethodContext resolvedKnnMethodContext = originalMappingParameters.getResolvedKnnMethodContext();
         VectorDataType vectorDataType = mappedFieldType.getVectorDataType();
 
-        final VectorSimilarityFunction vectorSimilarityFunction = knnMethodContext.getSpaceType()
+        final VectorSimilarityFunction vectorSimilarityFunction = resolvedKnnMethodContext.getSpaceType()
             .getKnnVectorSimilarityFunction()
             .getVectorSimilarityFunction();
 
         this.fieldType = vectorDataType.createKnnVectorFieldType(knnMappingConfig.getDimension(), vectorSimilarityFunction);
 
         if (this.hasDocValues) {
-            this.vectorFieldType = buildDocValuesFieldType(knnMethodContext.getKnnEngine());
+            this.vectorFieldType = buildDocValuesFieldType(resolvedKnnMethodContext.getKnnEngine());
```

Comment:
There is some ambiguity when to use resolved and when to use original. I am not sure if there is a way to resolve that, there is a comment around it, I think its best to add guidelines in that comment

Reply:
Sure, Ill add a comment